embedded/document/engine_test.go

```diff
@@ -1283,6 +1283,101 @@ func TestDocumentUpdate(t *testing.T) {
 		require.Len(t, revisions, 1)
 	})
 }
+func TestCustomDocumentID(t *testing.T) {
```

Comment:
add a negative tests, inserting document with the same id twice. Case of collisions of doc ids, if they are handled properly.

Reply:
good point